pkg/config/event_watcher.go

```diff
@@ -53,9 +54,22 @@ type EventWatcherReplacement struct {
 // LoadEventWatcher gives back parsed EventWatcher config after merging config files placed under
 // the .pipe directory. With "includes" and "excludes", you can filter the files included the result.
 // "excludes" are prioritized if both "excludes" and "includes" are given. ErrNotFound is returned if not found.
-func LoadEventWatcher(repoRoot string, includes, excludes []string) (*EventWatcherSpec, error) {
+func LoadEventWatcher(repoRoot string, includePatterns, excludePatterns []string) (*EventWatcherSpec, error) {
 	dir := filepath.Join(repoRoot, SharedConfigurationDirName)
-	files, err := ioutil.ReadDir(dir)
+
+	// Collect file paths recursively.
+	files := make([]string, 0)
+	err := filepath.Walk(dir,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() {
+				files = append(files, strings.TrimPrefix(path, dir+"/"))
+			}
+			return nil
+		},
+	)
```

Comment:
The number of files inside the `.pipe` directory is **expected** to not be large, so this is acceptable.